### PR TITLE
[Fix #63] Add support for definitions and schemas

### LINKED ISSUE #63: Support for definitions and schemas
Let users add definitions in code (can only be done now on sys.config) and add schemas to parameter fields.

### PR DESCRIPTION
**Pull request by harenson**

### CODE REVIEW

**Comment by elbrujohalcon** [comment]
@harenson can you add a little bit of documentation about this (if possible, with an example) to the Readme?

**Comment by harenson** [comment]
Done.

File: README.md [hunk 1]
```diff
@@ -138,5 +138,108 @@ Additionally, `cowboy_swagger` can be configured/customized from a `*.config` fi
...
+For adding definitions to your app, you have 2 choices:
+
+1. Add a `definitions` key to your cowboy_swagger `global_spec` map.
+2. Add them by calling `cowboy_swagger:add_definition/2` and send the
+   definition's name and properties.
+
+Let's say you want to describe a `POST` call to a `newspapers` endpoint that requires
+`name` and `description` fields only, you can do it like this:
+
+**Choice 1:**
```
Comment (elbrujohalcon) [inline README.md]: Choice -> Option

File: README.md [hunk 2]
```diff
@@ -138,5 +138,108 @@ Additionally, `cowboy_swagger` can be configured/customized from a `*.config` fi
...
+                }
+           }
+         }
+      }
+    ]
+  }
+]
+```
+
+**Choice2:**
```
Comment (elbrujohalcon) [inline README.md]: Choice -> Option

File: README.md [hunk 3]
```diff
@@ -138,5 +138,108 @@ Additionally, `cowboy_swagger` can be configured/customized from a `*.config` fi
...
+Now in your handler's trails callback function you can use it:
+
+```erlang
+...
+  RequestBody =
+    #{ name => <<"request body">>
+     , in => body
+     , description => <<"request body (as json)">>
+     , required => true
+     , schema => #{ % Note we are using the definitions here
```
Comment (elbrujohalcon) [inline README.md]: @harenson can we add an utility function to build this?
Something like…

```erlang
#{<<"$ref">> => <<"#/definitions/RequestBody">>} =
cowboy_swagger:schema(<<"RequestBody">>)
```

?

File: README.md [hunk 4]
```diff
@@ -138,5 +138,108 @@ Additionally, `cowboy_swagger` can be configured/customized from a `*.config` fi
...
+         , parameters => [RequestBody] % and then use that parameter here
+         }
+     },
+  Path = "/newspapers",
+  Options = #{path => Path},
+  [trails:trail(Path, newspapers_handler, Options, Metadata)].
+```
+
+What this does for you is add a nice `response`, `parameter` or `security`
+model in swagger-ui, so client developers can exactly know what parameters
```
Comment (elbrujohalcon) [inline README.md]: can exactly know => will know exactly

File: README.md [hunk 5]
```diff
@@ -138,5 +138,108 @@ Additionally, `cowboy_swagger` can be configured/customized from a `*.config` fi
...
+does the API expects for every endpoint.
```
Comment (elbrujohalcon) [inline README.md]: does the API expects -> the API expects

File: src/cowboy_swagger.erl [hunk 6]
```diff
@@ -60,6 +75,25 @@ to_json(Trails) ->
...
+add_definition(Name, Properties) ->
+  Definition = build_definition(Name, Properties),
+  CurrentSpec = application:get_env(cowboy_swagger, global_spec, #{}),
+  ExistingDefinitions = maps:get(definitions, CurrentSpec, #{}),
+  NewSpec = CurrentSpec#{definitions => maps:merge( ExistingDefinitions
+                                                  , Definition
+                                                  )},
+  application:set_env(cowboy_swagger, global_spec, NewSpec).
+
+-spec schema(DefinitionName::parameter_definition_name()) ->
```
Comment (gadgetci) [inline src/cowboy_swagger.erl]: According to **Dialyzer**:
>  Invalid type specification for function cowboy_swagger:schema/1. The success typing is (binary()) -> #{}

**Pull request merged**